graphs/lca.cpp

```diff
@@ -1,53 +1,64 @@
+/*
+ * Description: Lowest Common Ancestor (LCA) implemented with Binary Lifting.
+ * Supports functionality like kth Parent, distance between pair of nodes in
+ * number of edges, and distance between pair of nodes in sum of edge weight.
+ * Time: O(n * log(n)) for construction and all queries can be answered in
+ * O(log(n))
+ */
+
+
 struct lca {
+	typedef long long ll;
 	int Log;
 	vector<vector<int>> memo;
-	vector<int> depth;
-
-	lca(const vector<vector<int>> &graph, int root) {//0 - based nodes
-		int n = graph.size();
-		Log = 1;
-		while((1<<Log) < n) ++Log;
-		depth.resize(n,0);
-		memo.resize(n,vector<int>(Log));
-		dfs(root,root,graph);
-	}
+	vector<int> depth, dist;
 
-	void dfs(int node, int par, const vector<vector<int>> &graph) {
+	// use weights of 1 for unweighted tree
```

Comment:
Let's remove as we still have dist edges function

Reply:
Remove the DFS?

Reply:
remove the comment

Reply:
actually nevermind, let's keep it

Reply:
okay sounds good